fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -202,17 +202,22 @@ def to_s
       def boot
         return unless is_simulator
         return unless os_type == "iOS"
+        return if self.state == 'Booted'
 
         UI.message("Booting #{self}")
-        `xcrun simctl boot #{self.udid}` unless self.state == "Booted"
+
+        `xcrun simctl boot #{self.udid} 2>/dev/null`
+        self.state = 'Booted'
```

Comment:
How was this updated before after booting?

Reply:
Hi @janpio, are you asking how was this updated before I added the `boot` method? This method didn't exist, so simulators were booted using the [`Simulator.launch` method](https://github.com/fastlane/fastlane/pull/16833/files#diff-b6a3ba6dff446741cb76248e99156be7L47) before I added the `boot` method.

If you're asking, "why do we care what the state is", when I call the method that collects the logs for example, I may want to check, "is the device booted", so that I can avoid getting the error that occurs when getting logs from a shutdown simulator.

If you're asking, "how is this state not matching the actual device's state?", that's a different story. The `FastlaneCore::DeviceManager.simulators('iOS')` method can be used to create a _new_ `FastlaneCore::DeviceManager::Device` and that instance's shutdown method can be called to make the other state invalid. So, the new Device's state is shutdown, but the first Device's state is booted. I could have added code for the state to make it a method and _always_ query `simctl`, but that seemed potentially expensive, and nothing terrible seems to happen if I ignore the error, so that's why I put the `2>/dev/null` to the command.

Do any of those answer the question?

Reply:
I was not aware you added the `boot` method. So it this a follow up to an earlier PR?

Reply:
Yes, this a follow up change. I saw a problem when this code was being exercised and wanted to clean it up.

# 🧼